### PR TITLE
Ask if the user really wants to pastebin

### LINKED ISSUE #1027: pastebinning accidentally
A couple years ago at work, I clicked Pastebin --> dpaste.com accidentally, and it sent a CSS file I was working on to dpaste.com. This happened to be fine, because the file was served as-is on a web server, available to anyone who looks in the inspector. But it could have been a disaster, if I happened to work on closed-source code that has a severe security bug, for example.

When Tools --> Dpaste is clicked for the first time, it should ask whether you are sure you want to send the file to dpaste, and explain that it's basically public and someone might find it from there. It should also be possible to make it never ask again, either somewhere in the "are you sure" dialog or in Porcupine's setting dialog.

### PR DESCRIPTION
**Pull request by Akuli**
Fixes #1027

### CODE REVIEW

File: tests/test_pastebin_plugin.py [hunk 1]
```diff
@@ -217,3 +224,31 @@ def test_pasting_selected_indented_code(filetab, tabmanager, mocker, dont_run_in
     tabmanager.select(filetab)
     get_main_window().event_generate("<<Menubar:Pastebin/dpaste.com>>")
     mock_run.assert_called_once_with("bar\nif baz:\n    lol\n", PythonLexer)
+
+
+def test_are_you_sure_dialog(filetab, tmp_path, wait_until, mocker, monkeypatch):
+    mock_run = mocker.patch("porcupine.plugins.pastebin.DPaste.run")
+    mock_run.return_value = "https://foobar"
```
Comment (Akuli) [inline tests/test_pastebin_plugin.py:231]:
```suggestion
```

**Comment by Akuli** [comment]
@rdbende Can you review this? I added some padding to the labels, but I probably missed something else :)

File: porcupine/plugins/pastebin.py [hunk 2]
```diff
@@ -246,7 +247,66 @@ def pasting_done_callback(
         )
 
 
+def ask_are_you_sure(filename: str | None, paste_class: type[Paste]) -> bool:
+    window = tkinter.Toplevel()
```
Comment (rdbende) [inline porcupine/plugins/pastebin.py:251]:
```suggestion
    window = tkinter.Toplevel()
    window.transient(get_main_window())
```
Just so it doesn't appear as a separate icon on my taskbar ;)

File: porcupine/plugins/pastebin.py [hunk 3]
```diff
@@ -246,7 +247,66 @@ def pasting_done_callback(
...
+        result = True
+        window.destroy()
+
+    def no() -> None:
+        window.destroy()
+
+    button_frame = ttk.Frame(content)
+    button_frame.pack(fill="x")
+    ttk.Button(button_frame, text="Yes", command=yes).pack(side="left", expand=True)
+    ttk.Button(button_frame, text="No", command=no).pack(side="left", expand=True)
```
Comment (rdbende) [inline porcupine/plugins/pastebin.py:297]:
```suggestion
    ttk.Button(button_frame, text="Yes", command=yes).pack(side="left", expand=True, fill="x", padx=(0, 10))
    ttk.Button(button_frame, text="No", command=no).pack(side="left", expand=True, fill="x", padx=(10, 0))
```
bigger clickable area

File: porcupine/plugins/pastebin.py [hunk 4]
```diff
@@ -246,7 +247,66 @@ def pasting_done_callback(
...
+    window.title(f"Pastebin {filename}")
+
+    content = ttk.Frame(window, name="content", padding=10)
+    content.pack(fill="both", expand=True)
+    content.columnconfigure(0, weight=1)
+
+    label1 = ttk.Label(
+        content,
+        name="label1",
+        text=f"Do you want to send the content of {filename} to {paste_class.name}?",
```
Comment (rdbende) [inline porcupine/plugins/pastebin.py:261]: This text can be misleading, it says, `the content of {filename}` even if I just try to pastebin the selection.

**Review by rdbende — CHANGES_REQUESTED** [review]

**Comment by Akuli** [comment]
Thanks for the review!

**Pull request merged**